DuckDuckGoTests/MockSecureVault.swift

```diff
@@ -0,0 +1,119 @@
+//
+//  MockSecureVault.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import BrowserServicesKit
+
+final class MockSecureVault: SecureVault {
+
+    var storedAccounts: [SecureVaultModels.WebsiteAccount] = []
+    var storedCredentials: [Int64: SecureVaultModels.WebsiteCredentials] = [:]
+    var storedNotes: [SecureVaultModels.Note] = []
+    var storedIdentities: [SecureVaultModels.Identity] = []
+    var storedCards: [SecureVaultModels.CreditCard] = []
+
+    func authWith(password: Data) throws -> SecureVault {
+        return self
+    }
+
+    func resetL2Password(oldPassword: Data?, newPassword: Data) throws {}
+
+    func accounts() throws -> [SecureVaultModels.WebsiteAccount] {
+        return storedAccounts
+    }
+
+    func accountsFor(domain: String) throws -> [SecureVaultModels.WebsiteAccount] {
+        return storedAccounts.filter { $0.domain == domain }
```

Comment:
This might be a pain to update whenever the logic for secure vault changes. Given that it's not currently used in the tests (unless I missed it?), I might be tempted to ever comment something pointing this problem out, or turn it into just a more obvious no op

Reply:
(Although that said I would also love it if we didn't have that logic as tightly coupled to the DB, so it will depend on what you decided to do for the subdomain matching stuff)